window_manager/src/XWindow.cpp

```diff
@@ -200,4 +219,19 @@ void XWindow::stop_window_loop() {
         running_ = false;
     }
 }
+
+[[nodiscard]] const rendering::GLRendererPtr& XWindow::get_renderer() const {
+    return renderer_;
+}
+
+void XWindow::set_renderer(const rendering::GLRendererPtr& renderer_ptr) {
+    if (renderer_ptr == nullptr)
+        throw std::runtime_error("Null pointer assigned into renderer ptr");
+
+    renderer_ = renderer_ptr;
+    renderer_->on_stop([this](rendering::RenderingContextPtr ptr) {
```

Comment:
I am in doubt whether or not this one is needed.

Reply:
This is required in case, for some reason, renderer stops before window stops.